packages/playwright-test/src/dispatcher.ts

```diff
@@ -215,7 +216,12 @@ export class Dispatcher {
       }));
       result.status = params.status;
       test.expectedStatus = params.expectedStatus;
+      const inheritedAnnotations = test.annotations;
       test.annotations = params.annotations;
+      if (!this._inheritedAnnotations.has(test._id)) {
```

Comment:
Let's instead put the "inherited" annotations to a separate `test._inheritedAnnotations`.